### ISSUE TITLE
Unnecessary else clause (guard clause + assign and return)

### ISSUE THREAD

**Issue by marcoaaguiar** [sourcery-ai/sourcery#97]
### Issue description or question

This one seems a mix of guard clause + assign and return:
```python
def f(a=None):
    if a is None:
        return_val = 42
    else:
        # some long statement
        var = (i ** 2 for i in range(a))
        return_val = sum(var)

    return return_val
```

Could be refactored to
```python
def f(a=None):
    if a is None:
        return 42

    # some long statement
    var = (i ** 2 for i in range(a))
    return sum(var)
```

### Sourcery Version
0.8.2

### Code editor or IDE name and version

Discovered on NeoVim 0.5.0

### OS name and version

Ubuntu 20.04 via WSL

**Comment by Hellebore** [comment]
Looks like a nice change! Will look at adding it in.

**Comment by Hellebore** [comment]
These changes are in and will be in the next release!

**Comment by Hellebore** [comment]
Resolved with release of 0.8.3.
